mypy/errors.py

```diff
@@ -202,12 +202,17 @@ def add_error_info(self, info: ErrorInfo) -> None:
 
     def generate_unused_ignore_notes(self) -> None:
         for file, ignored_lines in self.ignored_lines.items():
-            for line in ignored_lines - self.used_ignored_lines[file]:
-                # Don't use report since add_error_info will ignore the error!
-                info = ErrorInfo(self.import_context(), file, None, None,
-                                 line, 'note', "unused 'type: ignore' comment",
-                                 False, False)
-                self.error_info.append(info)
+            if not self.is_typeshed_file(file):
+                for line in ignored_lines - self.used_ignored_lines[file]:
+                    # Don't use report since add_error_info will ignore the error!
+                    info = ErrorInfo(self.import_context(), file, None, None,
+                                    line, 'note', "unused 'type: ignore' comment",
+                                    False, False)
+                    self.error_info.append(info)
+
+    def is_typeshed_file(self, file):
```

Comment:
This function needs annotations.

If you make this the accepted, public interface to determine whether a file is in the typeshed or not, you can refactor the code in checker.py that sets is_typeshed_stub there like this:
```python
    self.is_typeshed_stub = self.errors.is_typeshed_file(path)
```